### DISCUSSION TITLE
jp2 compression

### DISCUSSION THREAD

**Discussion by Jmuccigr** [ImageMagick/ImageMagick#2833] [Help]
This is, I think, a quick script to generate 4 jp2 files at different compression rates:
```
for i in {1,2,10,100} \
do magick rose: -resize 1000% -format jp2 -define jp2:rate="$i" Downloads/rose_"$i".jp2 \
done
```
Here's the output of `magick -format "%f %C %Q %[quality] %b\n" info:` on those four output files:
```
rose_1.jp2 JPEG2000 92 0 169779B
rose_10.jp2 JPEG2000 92 0 96570B
rose_100.jp2 JPEG2000 92 0 9577B
rose_2.jp2 JPEG2000 92 0 169779B
```

Three questions:

1. Why is the quality 92 for all of them?
1. Why does %[quality] yield 0?
2. Why is the file with rate=2 the same size as rate=1 (uncompressed)?

MacOS 10.13.6
Version: ImageMagick 7.0.10-34 Q16 x86_64 2020-10-11 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1)
Delegates (built-in): bzlib fftw fontconfig freetype heic jng jp2 jpeg lcms ltdl lzma png tiff webp wmf x xml zlib

**Comment by urban-warrior** [comment]
ImageMagick defaults to quality 92.  Use the `-quality` option to specify a different compression quality.  Quality is zero when you query because the compression quality is not captured.  If you know how to capture the compression quality of a JPEG 2000-compressed image with the OpenJP2 delegate library, let us know and we will add support for the compression quality in a future release of ImageMagick.

**Reply by Jmuccigr** [reply]
I don’t, but can explore.

Can you explain why the size of the rate 2 file is the same as rate 1?
